modular_zubbers/code/_onclick/hud/screen_objects/hud_timer.dm

```diff
@@ -0,0 +1,213 @@
+
+/// UI obj holders for all your maptext needs
+/atom/movable/screen/text
+	name = null
+	icon = null
+	icon_state = null
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	screen_loc = "CENTER-7,CENTER-7"
+	maptext_height = 480
+	maptext_width = 480
+
+/// A screen object that shows the time left on a timer
+/atom/movable/screen/text/screen_timer
+	screen_loc = "CENTER-7,CENTER-7"
+	/// Left side of the HTML tag for maptext, style is also included
+	var/maptext_style_left = "<span class='maptext' style=font-size:6pt;text-align:center; align='top'>"
+	/// End tag of the HTML tag for maptext
+	var/maptext_style_right = "</span>"
+	/// The actual displayed content of the maptext, use ${timer}, and it'll be replaced with formatted time left
+	var/maptext_string
+	/// Timer ID that we're tracking, the time left of this is displayed as maptext
+	var/timer_id
+	/// The list of mobs in whose client.screens we are added to
+	var/list/timer_mobs = list()
+
+/atom/movable/screen/text/screen_timer/Initialize(
+		mapload,
+		list/mobs,
+		timer,
+		text,
+		offset_x = 150,
+		offset_y = -70,
+		style_start,
+		style_end,
+	)
+	. = ..(mapload, null)
+
+	if(!islist(mobs) && mobs)
+		mobs = list(mobs)
+	// Copy the list just in case the arguments list is a list we don't want to modify
+	if(length(mobs))
+		mobs = mobs.Copy()
+	if(!timer)
+		return INITIALIZE_HINT_QDEL
+	if(style_start)
+		maptext_style_left = style_start
+	if(style_end)
+		maptext_style_right = style_end
+	maptext_string = text
+	timer_id = timer
+	maptext_x = offset_x
+	maptext_y = offset_y
+	update_maptext()
+	if(length(mobs))
+		apply_to(mobs)
+
+/atom/movable/screen/text/screen_timer/process()
+	if(!timeleft(timer_id))
+		delete_self()
+		return
+	update_maptext()
+
+/// Adds the object to the client.screen of all mobs in the list, and registers the needed signals
+/// TODO: WEAKREFS ARE FUCKY WITH REMOVING
```

Comment:
Given this comment and a bug report in #bug-reports, I suspect that screen timers are causing the server to crash.

Reply:
It seems to be an incorrect DB query as TG code doesnt just crash that easily internally, but can very easily crash on external calls

Reply:
@BurgerLUA link to bug report?

Reply:
Found it, https://discord.com/channels/1059199070016655462/1061433759947313233/1270651040718389260

Reply:
The crashing was calling initial on get_power_explanation_extended(), in get_power_explanation(), in most other procs i had overriden this, but this would cause it to get called when the antag info window automatically opens when a vassal is vassalized, and it wasn't overriden on the damn vassal power.